spec/rubocop/cop/internal_affairs/redundant_described_class_as_subject_spec.rb

```diff
@@ -62,4 +62,16 @@
       end
     RUBY
   end
+
+  it 'registers an offense without `describe` DSL' do
+    expect_offense(<<~RUBY)
+      shared_context "shared cop context" do
+        subject(:cop) { described_class.new }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Remove the redundant `subject` and specify `:config` in `describe`.
```

Comment:
Shouldn't this case be allowed since it depends on the context?

Reply:
Hm, that might be so. I'm not sure, I'm not really familiar with this topic. Are you saying there should be no offenses at all?

Reply:
Rephrasing the question, why isn't it automatically removed during autocorrection?

Reply:
I don't know if there's a reason to keep it. [Here's](https://github.com/anycable/anycable-rails/blob/934f1bcd47764cf95c05172899c91554af48fa73/spec/cops_spec_helper.rb#L14) where I encountered the error

Reply:
I guess we can just remove this during the autocorrection

Reply:
In that case, it would be more appropriate to use `expect_correction` instead of `expect_no_corrections`.

Reply:
Alright, fixed